core/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala

```diff
@@ -53,11 +53,15 @@ import org.apache.spark.util.Utils
  * @param size the number of actions in the checkpoint
  * @param parts the number of parts when the checkpoint has multiple parts. None if this is a
  *              singular checkpoint
+ * @param sizeInBytes the number of bytes of the checkpoint.
+ * @param numOfAddFiles the number of AddFile actions in the checkpoint
  */
 case class CheckpointMetaData(
     version: Long,
     size: Long,
-    parts: Option[Int])
+    parts: Option[Int],
+    sizeInBytes: Long,
+    numOfAddFiles: Long)
```

Comment:
Should these new fields be `Option[Long]` rather than `Long`, for backward compat reasons? Or should we just continue relying on the JSON reader to default-fill missing fields when reading an older last checkpoint file?

Reply:
Could you add `@JsonDeserialize(contentAs = classOf[java.lang.Long])`? `Option[Long]` will not work by default without `JsonDeserialize`. This is similar to https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/actions/actions.scala#L219-L220